### DISCUSSION TITLE
사용자 주소 값을 어떻게 관리할까요?

### DISCUSSION THREAD

**Discussion by clyde-yoon** [team-final/albaform#93] [General]
# 상황

피그마에선,
![image](https://github.com/user-attachments/assets/bdc5e387-fb50-4b06-b60e-8e7196375f5d)
다음의 사진 같이 사장님의 가게 위치를 input 하나로 받고있습니다. 
![image](https://github.com/user-attachments/assets/cec57830-661f-49d5-a3bd-a4d23cbad94e)
알바폼 상세보기에는 사장님의 가게의 해당 주소가 적혀져있습니다.

사실상 지도에 마크업을 하기위해서 필요한 부분은 
'서울특별시 중구 청계천로 100' 까지여도 충분합니다. 

# 문제
사용자(사장님)는 가게 위치에 대한 정확한 주소를 입력해야합니다. 이는 해당위치를 좌표로 변환시켜 알바폼 상세보기의 지도에 필요하기 때문입니다. 그래서 사용자의 위치를 정확하게 입력하도록 강제해야하는데,
 
![image](https://github.com/user-attachments/assets/f4f9083b-5dca-482d-a5b3-55ee5934e50c)

다음 과 같은 창을 통해 정확환 도로명주소를 받을 수 있습니다.
(아마 한번쯤은 다들 보셨을 것입니다.)

그런데 이렇게 위치를 받으면 사실상 '도로명' 에 해당하는 정보만 가집니다. 더 자세히 말하면 위치는 알지만 몇층인지, 몇호인지는 더 정확한 값을 사용자에게 요구해야합니다. 

그러나 상세주소까지 받아내면, 도로명 주소를 통해 좌표로 변환하는 과정에서 문제가 생길 수 있습니다. 

geocoder.addressSearch 함수는 도로명 주소나 지번 주소를 기준으로 변환하기 때문입니다.

![image](https://github.com/user-attachments/assets/70ac54b3-80c1-437a-8814-8c16fb6da80e)

그래서 현재의 상황에서 상세주소를 어떻게 저장하고 다룰 것인지, 어떻게 필터링을 할것인지 대한 논의가 필요할 것 같습니다.

**Comment by clyde-yoon** [comment]
일단은 반드시 상세주소까지 받아야합니다. 

그래서 제가 생각해본거는
정보입력 때 가게위치 아래에 상세 주소를 담을 새로운 input태그를 만들고 상세주소를 입력까지 마치고 나면 이 두 input의 vlaue를 하나로 합치는데, 이때 상세주소 앞에는 문자열 / 를 더합니다. 그러면 서버에 보낼 때 
서울특별시 중구 청계천로 100/동관 1층 코드잇 스터디카페
이렇게 될것이고, 좌표에서 필요한 부분은 / 전까지만 추출해서 구하게 하고, 주소를 나타낼떄는 배열함수 돌려서 / 부분을 제거한 새로운 문자열 만들어서 나타내는 건 어떨까요?

그러나 이렇게 구현한다면, 상세보기 input에는 / 문자를 사용하지 못하게 막아야하는 추가적인 기능이 더 필요한 불편한 상황도 생길 수 있습니다.

**Reply by clyde-yoon** [reply]
![image](https://github.com/user-attachments/assets/ee40fc87-a1b2-4a03-bdb9-f6776de77d27)

(현재 구현은 되어있습니다 토의 끝나고 pr올리겠습니다)

**Comment by JiinSeok** [comment]
공감합니다. 지도 앱이 사용할 정보인데 지도 앱이 이해할 수 없는 형태로 저장해서는 안 되겠죠. (주소 찾기 팝업으로 있는 주소만 선택할 수 있게 강제, 상세주소 안 붙이기!!) 지금은 지도에만 쓰지만, 상세주소 따로 저장 못해서 주소 정보 규격화 못하면 지역별/우리동네 알바폼 보기 같은 확장도 불가능합니다. (아아 정보가 있는데 써먹질 못하다니...)
그렇다고 상세주소를 빼고 저장하면 이 서비스의 모든 이용자가 사용할 정보를 제외하는 것이니 서비스 취지에 맞지 않습니다. 알바생이 어디로 찾아갈지 매번 개별 연락하라는 뜻이니까요.

별도로 저장할 수 있게 location을 string 아닌 객체(`{address: string, detailAddress: string}`)로 바꿔달라고 요청하면 좋을 것 같습니다. 
불가능하다고 답이 오면 윤재님이 제안하신 기준 문자로 합치고/슬라이싱하는 게 제 생각에도 최선입니다(다른 메모할 곳도 없고). 다만 /보다는 1) 동일한 상황에서 많이 쓰이는 ,로 하거나 2) 거의 쓰이지 않을 [대괄호로 감싸기] 등이 좀 더 좋을 것 같습니다. 특정 문자를 막는 건 정규표현식 설정으로 얼마든지 가능하고 전혀 복잡하지 않으니 걱정하실 것 없다고 생각합니다 !!!!

**Reply by clyde-yoon** [reply]
오호 바꿔달라는 건 생각을 못했네요 수정요청해볼까요?

**Reply by JiinSeok** [reply]
@clyde-yoon 네!!! 이건 당연한 거라고 생각합니다 !!!! 지도 앱 쓰는 기획인데 지도 앱에 들어갈 주소랑 아닌 주소 따로 다룰 수 있어야죠!

**Comment by JiinSeok** [comment]
타입이 string이니까 모양이 string인 JSON을 넣고, 그걸 JSON.parse 해서 객체로 쓰자.
그러면 객체로 주소, 상세주소를 나누어 저장활 수 있을 것.
